packages/cozy-client/src/CozyClient.js

```diff
@@ -1195,6 +1196,10 @@ instantiation of the client.`
       this.dispatch(receiveQueryResult(null, { data }))
     })
   }
+
+  toJSON() {
+    return new SnapshotClient({ uri: this.options.uri })
```

Comment:
Not blocking, but why return a `SnapshotObject ` and not plain JSON? This would also be more in line with the expected behavior of `toJSON` even outside of tests, and even if the JSON representation is incomplete.

Reply:
Becuase I want to have `CozyClient { uri: 'https://...' }` in the tests and not `Object { uri: 'https:///' }`. `pretty-format` supports having class instances returned inside toJSON() since it looks at the constructor name (which actually should not return JSON but POJO).

Reply:
I tested that JSON.stringify() worked as expected even if an instance of a custom class was returned via toJSON and it works as expected.

```
class B { constructor (attrs) { Object.assign(this, attrs) } }

class A { toJSON() { return new B({ uri: 'abc' }) } }
undefined
a = new A()
Object {  }

JSON.stringify(a)
"{\"uri\":\"abc\"}"
```